workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/androidx/KeyedStateRegistryOwner.kt

```diff
@@ -0,0 +1,33 @@
+package com.squareup.workflow1.ui.androidx
+
+import androidx.lifecycle.LifecycleOwner
+import androidx.savedstate.SavedStateRegistry
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import androidx.savedstate.ViewTreeSavedStateRegistryOwner
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * The implementation of [SavedStateRegistryOwner] that should be installed on every immediate
+ * child view of container root views (e.g. content views, e.g. backstack frames) so that when
+ * a view inside a container calls [ViewTreeSavedStateRegistryOwner.get] on itself, one of these
+ * is returned.
+ *
+ * The container should use a [StateRegistryAggregator] to manage its set of
+ * [KeyedStateRegistryOwner] instances, which will save and restore them
+ * via its own [ViewTreeSavedStateRegistryOwner].
+ *
+ * To create an instance, call [StateRegistryAggregator.installChildSavedStateRegistryControllerOn].
+ *
+ * @param key The key used to save and restore this controller from a [SavedStateRegistry].
+ * @param lifecycleOwner The [LifecycleOwner] that will be delegated to by this instance. Note that
+ * [SavedStateRegistryOwner] extends [LifecycleOwner].
```

Comment:
Yeah, I'm not sure that this is useful information (and I think it was a mistake that the androidx team made this choice, can't see this redundant lifecycle is useful, or even safe to rely on). I'll delete the comment.

Reply:
Actually, this _is_ kind of useful, if only to explain why the hell we're doing this at all. Rephrased it a bit:

```
 * @param lifecycleOwner The [LifecycleOwner] that will be delegated to by this instance.
 * (Required because [SavedStateRegistryOwner] extends [LifecycleOwner] for no clear reason.)
```